src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -138,6 +160,27 @@ public override void OnRemoveBehavior()
 
         Dictionary<int, CrpgPeer> crpgPeerByUserId = new();
         List<CrpgUserUpdate> userUpdates = new();
+        Dictionary<CrpgPeer, IList<CrpgUserDamagedItem>> brokenItems = new();
+
+        Dictionary<int, int> repairCostByCrpgUserId = new();
+        foreach (NetworkCommunicator networkPeer in networkPeers)
+        {
+            var missionPeer = networkPeer.GetComponent<MissionPeer>();
+            var crpgPeer = networkPeer.GetComponent<CrpgPeer>();
+            if (missionPeer == null || crpgPeer?.User == null)
+            {
+                continue;
+            }
+
+            if (missionPeer.Team != null && missionPeer.Team.Side != BattleSideEnum.None)

```

Comment:
I would prefer if it would stay like this in this PR (as i didn't change the logic but rather splitted the `BreakItems` part from the other userUpdate calculations.
I could make a separate PR for the Reward/Upkeep fix before spawning in (This is the problem right?)
This would keep things clean and wouldn't implement a bugfix with a new system in the same commit.